src/libltfs/index_criteria.c

```diff
@@ -288,27 +288,38 @@ int index_criteria_parse_name(const char *criteria, size_t len, struct index_cri
 	/* Assign rules to the glob_patterns[] array */
 	rule = rule+5;
 	for (delim = rule; *delim; delim++) {
+		bool do_delim_assign = false;
+		bool do_rule_add = false;
 		if (*delim == ':') {
-			*delim = '\0';
-			rule_ptr->percent_encode = fs_is_percent_encode_required(rule);
-			rule_ptr->name = strdup(rule);
-			rule_ptr++;
-			rule = delim+1;
-		} else if (*delim == '/') {
-			*delim = '\0';
-			rule_ptr->percent_encode = fs_is_percent_encode_required(rule);
-			rule_ptr->name = strdup(rule);
-			rule_ptr++;
-		} else if (*(delim+1) == '\0') {
-			rule_ptr->percent_encode = fs_is_percent_encode_required(rule);
-			rule_ptr->name = strdup(rule);
-			rule_ptr++;
+			do_delim_assign = true;
+			do_rule_add = true;
 		}
+		else if (*delim == '/') {
```

Comment:
Please follow the convention. You must not break a line before `else`.

It looks there are multiple points that is same style. Please correct all of them.